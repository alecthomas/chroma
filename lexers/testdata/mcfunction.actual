fill 29 4 26 ~ ~ ~ minecraft:air replace minecraft:water

# NBT (JSON-like) value.
summon zombie ~ ~ ~ {NoAI:1b}

# Target selector
gamemode creative @a

# With argument
kill @e[type=chicken,nested={a0=0,a1=1}]

# Ragne
kill @e[y_rotation=0..180]
