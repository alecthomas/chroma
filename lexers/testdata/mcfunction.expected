[
  {"type":"KeywordReserved","value":"fill"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberInteger","value":"29"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberInteger","value":"4"},
  {"type":"TextWhitespace","value":" "},
  {"type":"LiteralNumberInteger","value":"26"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"~"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"~"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"~"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"minecraft:air"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"replace"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"minecraft:water"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"CommentSingle","value":"# NBT (JSON-like) value.\n"},
  {"type":"KeywordReserved","value":"summon"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"zombie"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"~"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"~"},
  {"type":"TextWhitespace","value":" "},
  {"type":"NameBuiltin","value":"~"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Punctuation","value":"{"},
  {"type":"NameTag","value":"NoAI"},
  {"type":"Punctuation","value":":"},
  {"type":"LiteralNumber","value":"1b"},
  {"type":"Punctuation","value":"}"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"CommentSingle","value":"# Target selector\n"},
  {"type":"KeywordReserved","value":"gamemode"},
  {"type":"TextWhitespace","value":" "},
  {"type":"Text","value":"creative"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordConstant","value":"@a"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"CommentSingle","value":"# With argument\n"},
  {"type":"KeywordReserved","value":"kill"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordConstant","value":"@e"},
  {"type":"Punctuation","value":"["},
  {"type":"NameAttribute","value":"type"},
  {"type":"Punctuation","value":"="},
  {"type":"Text","value":"chicken"},
  {"type":"Punctuation","value":","},
  {"type":"NameAttribute","value":"nested"},
  {"type":"Punctuation","value":"={"},
  {"type":"NameAttribute","value":"a0"},
  {"type":"Punctuation","value":"="},
  {"type":"LiteralNumberInteger","value":"0"},
  {"type":"Punctuation","value":","},
  {"type":"NameAttribute","value":"a1"},
  {"type":"Punctuation","value":"="},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Punctuation","value":"}]"},
  {"type":"TextWhitespace","value":"\n\n"},
  {"type":"CommentSingle","value":"# Ragne\n"},
  {"type":"KeywordReserved","value":"kill"},
  {"type":"TextWhitespace","value":" "},
  {"type":"KeywordConstant","value":"@e"},
  {"type":"Punctuation","value":"["},
  {"type":"NameAttribute","value":"y_rotation"},
  {"type":"Punctuation","value":"="},
  {"type":"LiteralNumberInteger","value":"0"},
  {"type":"Punctuation","value":".."},
  {"type":"LiteralNumberInteger","value":"180"},
  {"type":"Punctuation","value":"]"},
  {"type":"TextWhitespace","value":"\n"}
]
